map/discovery/discovery_manager.hpp

```diff
@@ -0,0 +1,137 @@
+#pragma once
+
+#include "search/city_finder.hpp"
+
+#include "map/discovery/discovery_client_params.hpp"
+#include "map/search_api.hpp"
+
+#include "partners_api/booking_api.hpp"
+#include "partners_api/locals_api.hpp"
+#include "partners_api/viator_api.hpp"
+
+#include "platform/platform.hpp"
+
+#include "indexer/index.hpp"
+
+#include "geometry/point2d.hpp"
+#include "geometry/rect2d.hpp"
+
+#include "base/thread_checker.hpp"
+
+#include <functional>
+#include <string>
+#include <vector>
+
+namespace discovery
+{
+class Manager final
+{
+public:
+  struct APIs
+  {
+    APIs(SearchAPI & search, viator::Api const & viator, locals::Api & locals)
+      : m_search(search), m_viator(viator), m_locals(locals)
+    {
+    }
+
+    SearchAPI & m_search;
+    viator::Api const & m_viator;
+    locals::Api & m_locals;
+  };
+
+  Manager(Index const & index, search::CityFinder & cityFinder, APIs const & apis);
+
+  struct Params
```

Comment:
All typedefs and structs must be before ctors - https://google.github.io/styleguide/cppguide.html#Declaration_Order

Reply:
ok